distributed/tests/test_speculative.py

```diff
@@ -0,0 +1,93 @@
+from dask import delayed
+
+from distributed.utils_test import gen_cluster, inc, dec
+
+
+@gen_cluster(client=True, config={"dask.optimization.fuse.active": False})
+async def test_speculative_assignment_simple(c, s, a, b):
+    x = delayed(inc)(1)
+    y = delayed(inc)(x)
+    z = delayed(dec)(y)
+
+    zz = c.compute(z)
+    result = await zz
+    assert result == 2
+    assert (y.key, "speculative", "ready") in a.story(y.key)
+    assert (z.key, "speculative", "ready") in a.story(z.key)
```

Comment:
Another way to test this would be to have a slowinc, followed by two fast-incs.  Once the slow-inc is found on the worker we assert that all three tasks are in `worker.tasks`.  This seems like a test that is slightly less likely to go bad as people change around internals.  (what we have here is great too though, I'm just spit-balling)